app/controllers/contributions_controller.rb

```diff
@@ -1,4 +1,17 @@
 class ContributionsController < ApplicationController
+  def new
+    @contribution = Contribution.new
+  end
+
+  def create
+    @contribution = Contribution.new
+    @contribution.user = current_user # user does not own this project, user is logged in
+    @contribution.project = Project.find(params[:id])
```

Comment:
if you are going to reuse line 9, keep it in private method. even in edit and do before_action at the top.